libs/os/src/arch/sim/os_arch_sim.c

```diff
@@ -179,18 +196,53 @@ os_arch_in_critical(void)
 }
 
 void
-os_arch_idle(void)
+os_tick_idle(os_time_t ticks)
 {
+    int rc;
+    struct itimerval it;
+
+    OS_ASSERT_CRITICAL();
+
+    if (ticks > 0) {
+        /*
+         * Enter tickless regime and set the timer to fire after 'ticks'
+         * worth of time has elapsed.
+         */
+        it.it_value.tv_sec = ticks / OS_TICKS_PER_SEC;
+        it.it_value.tv_usec = (ticks % OS_TICKS_PER_SEC) * OS_USEC_PER_TICK;
```

Comment:
I would do this (ticks * OS_USEC_PER_TICK) % OS_TICKS_PER_SEC. But this change is not a must.